node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNodeTest.java

```diff
@@ -18,32 +18,132 @@
+                node("4", node(4, 4), host(21, 22), host(40, 40)),
+                node("5", node(4, 4), host(21, 21), host(40, 80))
+        );
+        assertOrder(expected);
+    }
+
+    @Test
+    public void testOrderingByAllocationSkew2() {
+        // The same as testOrderingByAllocationSkew1, but deviating from mean (20) in the other direction.
+        // Since we don't choose the node with the lowest skew, but with the largest skew *reduction*
+        // this causes the opposite order.
+        List<PrioritizableNode> expected = List.of(
+                node("4", node(4, 4), host(19, 18), host(40, 40)),
+                node("3", node(4, 4), host(18, 20), host(40, 40)),
+                node("2", node(4, 4), host(19, 20), host(40, 40)),
+                node("1", node(4, 4), host(20, 20), host(40, 40)),
+                node("5", node(4, 4), host(19, 19), host(40, 80))
+        );
+        assertOrder(expected);
+    }
 
+    @Test
+    public void testOrderingByAllocationSkew3() {
+        // The same as testOrderingByAllocationSkew1, but allocating skewed towards cpu
+        List<PrioritizableNode> expected = List.of(
+                node("1", node(4, 2), host(20, 20), host(40, 40)),
+                node("2", node(4, 2), host(21, 20), host(40, 40)),
+                node("4", node(4, 2), host(21, 22), host(40, 40)),
+                node("3", node(4, 2), host(22, 20), host(40, 40)),
+                node("5", node(4, 2), host(21, 21), host(40, 80))
+        );
+        assertOrder(expected);
+    }
+
+    @Test
+    public void testOrderingByAllocationSkew4() {
+        // The same as testOrderingByAllocationSkew1, but allocating skewed towards memory
+        List<PrioritizableNode> expected = List.of(
+                node("5", node(2, 10), host(21, 21), host(40, 80)),
+                node("3", node(2, 10), host(22, 20), host(40, 40)),
+                node("2", node(2, 10), host(21, 20), host(40, 40)),
+                node("1", node(2, 10), host(20, 20), host(40, 40)),
+                node("4", node(2, 10), host(21, 22), host(40, 40))
+        );
+        assertOrder(expected);
+    }
+
+    @Test
+    public void testOrderingByAllocationSkew5() {
+        // node1 is skewed towards cpu (without this allocation), allocation is skewed towards memory, therefore
+        // node 1 is preferred (even though it is still most skewed)
+        List<PrioritizableNode> expected = List.of(
+                node("1", node(1, 5), host(21, 10), host(40, 40)),
+                node("2", node(1, 5), host(21, 20), host(40, 40)),
+                node("3", node(1, 5), host(20, 20), host(40, 40)),
+                node("4", node(1, 5), host(20, 22), host(40, 40))
+        );
+        assertOrder(expected);
+    }
+
+    private void assertOrder(List<PrioritizableNode> expected) {
         List<PrioritizableNode> copy = new ArrayList<>(expected);
         Collections.shuffle(copy);
```

Comment:
Remove? Shuffle and then sort...

Reply:
It's to test the order resulting from sort. Before shuffle it is already sorted correctly.